metron-analytics/metron-profiler/src/main/java/org/apache/metron/profiler/bolt/ProfileBuilderBolt.java

```diff
@@ -243,8 +244,18 @@ private void flush(Tuple tickTuple) {
     measurement.setValue(result);
     emit(measurement, tickTuple);
 
+    // Execute the update with the old state
+    Map<String, String> tickUpdate = profileConfig.getTickUpdate();
+    Map<String, Object> state = executor.getState();
+    if(tickUpdate != null) {
+      tickUpdate.forEach((var, expr) -> executor.assign(var, expr, executor.getState()));
+    }
     // clear the execution state to prepare for the next window
     executor.clearState();
+    //make sure that we bring along the update state
+    if(tickUpdate != null) {
+      tickUpdate.forEach((var, expr) -> executor.getState().put(var, state.get(var)));
```

Comment:
We are taking on risk of premature worker death for one tick, just like we are currently in the profiler.  If the worker gets respawned, it will rerun init, which should initialize the state appropriately (with the lookback window intact, but with the current tick's work of data lost).  In the case of MAD, init and tickUpdate are identical sets of expressions.

We really should make sure we spell out in the docs that worker death will cause state that has not been persisted to be lost.  I'm not sure if it is or not, honestly.